### PR TITLE
Escape variables before being embedded into queries

### LINKED ISSUE #44: Schema Importer fails if specified namespace contains the reserved keywords of PostgreSQL

### PR DESCRIPTION
**Pull request by choplin**
## Description

This PR fixes the bug reported in #44 that if names of ScalarDB objects, including namespace, table, etc.,  are invalid on the PostgreSQL side (e.g., PostgreSQL keywords), Schema Importer fails due to the invalid query.

## Related issues and/or PRs

Closes #44

## Changes made

This PR escapes all variables that are embedded into the queries. More precisely, the following two methods provided by the PostgreSQL JDBC driver are used to escape variables based on how the variable is used in the query:
- [escapeIdentifier](https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/core/Utils.html#escapeIdentifier-java.lang.StringBuilder-java.lang.String-)
- [escapeLiteral](https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/core/Utils.html#escapeLiteral-java.lang.StringBuilder-java.lang.String-boolean-)

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

None

## Release notes

- Fixed a bug that caused Schema Importer to fail when ScalarDB objects include PostgreSQL keywords

### CODE REVIEW

File: schema-importer/app/src/main/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/DatabaseContext.kt [hunk 1]
```diff
@@ -42,3 +43,10 @@ fun executeUpdateWithLogging(stmt: Statement, logger: KLogger, sql: String) {
     logger.debug { sql }
     stmt.executeUpdate(sql)
 }
+
+fun escapeIdentifier(ident: String): String {
+    return Utils.escapeIdentifier(null, ident).toString()
+}
+fun escapeLiteral(literal: String): String {
```
Comment (komamitsu) [inline schema-importer/app/src/main/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/DatabaseContext.kt:50]: An empty line before this method would be nice.

File: schema-importer/app/src/test/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/CreateSchemaTest.kt [hunk 2]
```diff
@@ -49,8 +49,8 @@ class CreateSchemaTest {
         CreateSchema(ctx, namespaces).run()
 
         verify {
-            statement.executeUpdate("CREATE SCHEMA IF NOT EXISTS ns1;")
-            statement.executeUpdate("CREATE SCHEMA IF NOT EXISTS ns2;")
+            statement.executeUpdate("CREATE SCHEMA IF NOT EXISTS \"ns1\";")
```
Comment (komamitsu) [inline schema-importer/app/src/test/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/CreateSchemaTest.kt:52]: [trivial] Maybe we don't need the escape? But I'm not sure which is better.
```suggestion
            statement.executeUpdate("""CREATE SCHEMA IF NOT EXISTS "ns1";""")
```

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!

---

FYI: I was able to load some objects whose names are `reserved words` (e.g., `order`) in my local test environment.

```sql
scalardb=# select schemaname, viewname from pg_views where schemaname in ('customer', 'order');
 schemaname |  viewname
------------+------------
 customer   | customers
 order      | orders
 order      | statements
 order      | items
(4 rows)
```

**Comment by brfrn169** [comment]
LGTM! Thank you!

**Review by brfrn169 — APPROVED** [review]

File: schema-importer/app/src/main/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/DatabaseContext.kt (hunk 1 above)
Comment (choplin) [reply schema-importer/app/src/main/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/DatabaseContext.kt:51]: Thanks! I added the blank line in https://github.com/scalar-labs/scalardb-analytics-postgresql/pull/45/commits/9f87de662447482935ff076f39acee5be7ea18ac

File: schema-importer/app/src/test/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/CreateSchemaTest.kt (hunk 2 above)
Comment (choplin) [reply schema-importer/app/src/test/kotlin/com/scalar/db/analytics/postgresql/schemaimporter/CreateSchemaTest.kt]: I see. I changed to use the triple double-quotes in https://github.com/scalar-labs/scalardb-analytics-postgresql/pull/45/commits/9f87de662447482935ff076f39acee5be7ea18ac.

**Pull request merged**